docs/design/node_taint_eviction.md

```diff
@@ -2,36 +2,42 @@
-The actual replacement of all Pods running on tainted nodes, depends on the data that must be replicated and the number of affected Pods.
+The goal of this design is to implement a mechanism that allows the operator to replace Pods that are running on Nodes with a given set of taints.
+It's expected that the operator will detect the Pods running on tainted Nodes in a timely manner, either by receiving an event triggering the reconciliation loop or by periodic reconciliation events that happen every 10h by default. Once the operator marks Pods on tainted Nodes as to-be-replaced, the operator will re-use the replacement logic to replace those Pods. The actual replacement speed depends on the speed of excluding a Pods, the number of Pods to replace, and the allowed maximum number of inflight replacements (defined by `automationOptions.replacements.maxConcurrentReplacements`).
 
 ## Current Implementation
+The operator does not have this feature. It does not have access to list Node information. It does not consider taint-related events in Pod's condition.
 
-We don't have a current implementation of this and the operator is currently not able to access any node information.
-The closest implementation to this is the [Replace Failed ProcessGroups reconciler](https://github.com/FoundationDB/fdb-kubernetes-operator/blob/main/controllers/replace_failed_process_groups.go), which could be extended for this.
+The operator does replace failed ProcessGroups in the [Replace Failed ProcessGroups reconciler](https://github.com/FoundationDB/fdb-kubernetes-operator/blob/main/controllers/replace_failed_process_groups.go). It can extend it to support this feature.
 
 ## Proposed Design
 
-We have to make sure that this feature can be enabled and disabled, e.g. if someone wants to run the operator inside a cluster without node access.
-The `AutomaticReplacementOptions` will be extended to support the configuration of replacements for Pods that run on tainted nodes.
-The new field will be called `taints` and will contain a list, the default would be an empty list, of taint configurations.
-That configuration will include the taint key, this has to be an exact match or a wildcard `*`, and the duration this taint must be present on the node.
-The `durationInSeconds` allow to specify how sensitive the operator should be to taints.
-If a wildcard is specified as key this will be used as default value, except there is an exact match defined for a taint key.
+### Design Requirements
+**Requirement 1.** The feature can be enabled and disabled by users, because (1) when someone wants to run the operator inside a k8s cluster without Node access, they should be able to bypass this feature; (2) when we notice problems during roll-out of this feature, we need to disable it online.
+
+The feature can be disabled by option `automationOptions.replacements.taintReplacementOptions = {}`.
```

Comment:
This is not a requirement but an implementation detail, this should be in the user manual.

Reply:
This is the API definition/design for users.
During our discussion in implementation PR, we debated on "should we support using {key=*, duration=-1}" to disable the entire taint feature.  If the design defines it clearly,  the implementation will have less ambiguity.

Reply:
If this is intended for users we should make this example in `yaml` in the context of the `FoundationDBClusterSpec` and not a one line go code example as most (if not all) users will be creating and managing clusters using yaml files with the `FoundationDBCluster` definition.

Reply:
add a yaml example